mobile-app/app/screens/AppNavigator/screens/Balances/screens/WithdrawFutureSwapScreen.tsx

```diff
@@ -0,0 +1,332 @@
+import { useEffect, useMemo, useState } from 'react'
+import { Platform, StyleProp, ViewStyle } from 'react-native'
+import BigNumber from 'bignumber.js'
+import Slider from '@react-native-community/slider'
+import { StackScreenProps } from '@react-navigation/stack'
+import { tailwind } from '@tailwind'
+import { translate } from '@translations'
+import { View } from '@components'
+import { getNativeIcon } from '@components/icons/assets'
+import { ThemedIcon, ThemedScrollView, ThemedText, ThemedTouchableOpacity, ThemedView } from '@components/themed'
+import { WalletTextInput } from '@components/WalletTextInput'
+import { BalanceParamList } from '../BalancesNavigator'
+import { NumberRow } from '@components/NumberRow'
+import { useTokenPrice } from '../hooks/TokenPrice'
+import { InfoRow, InfoType } from '@components/InfoRow'
+import { useWhaleApiClient } from '@shared-contexts/WhaleContext'
+import { useLogger } from '@shared-contexts/NativeLoggingProvider'
+import { SubmitButtonGroup } from '@components/SubmitButtonGroup'
+import { hasTxQueued } from '@store/transaction_queue'
+import { hasTxQueued as hasBroadcastQueued } from '@store/ocean'
+import { RootState } from '@store'
+import { useSelector } from 'react-redux'
+import { NavigationProp, useNavigation } from '@react-navigation/native'
+import { useFutureSwapDate } from '../../Dex/hook/FutureSwap'
+
+type Props = StackScreenProps<BalanceParamList, 'WithdrawFutureSwapScreen'>
+
+export function WithdrawFutureSwapScreen (props: Props): JSX.Element {
+  const { futureSwap: { source, destination } } = props.route.params
+  const navigation = useNavigation<NavigationProp<BalanceParamList>>()
+  const client = useWhaleApiClient()
+  const logger = useLogger()
+  const { getTokenPrice } = useTokenPrice()
+  const hasPendingJob = useSelector((state: RootState) => hasTxQueued(state.transactionQueue))
+  const hasPendingBroadcastJob = useSelector((state: RootState) => hasBroadcastQueued(state.ocean))
+  const [amountToWithdraw, setAmountToWithdraw] = useState('0')
+  const [fee, setFee] = useState<BigNumber>(new BigNumber(0.0001))
+  const blockCount = useSelector((state: RootState) => state.block.count ?? 0)
+  const executionBlock = useSelector((state: RootState) => state.futureSwaps.executionBlock)
+  const { isEnded } = useFutureSwapDate(executionBlock, blockCount)
```

Comment:
we should snapshot the `isEnded` flag from the previous screen, because execution block will be renewed once it is reached, same thing for confirm withdraw screen

Reply:
Updated!

Reply:
@pierregee actually i was wrong, we should snapshot `executionBlock` instead, so we can disable the button by block count reaches that value